civis/tests/test_io.py

```diff
@@ -536,102 +536,119 @@ def test_civis_file_to_table_table_does_not_exist_some_sql_types_missing(
-                                              polling_interval=None)
-
-        m_run_cleaning.assert_called_once_with(
-            [mock_file_id], self.mock_client, False, True, 'comma', True
-        )
-        m_process_cleaning_results.assert_called_once_with(
-            [mock.sentinel.cleaning_future],
-            self.mock_client,
-            True,
-            False,
-            'comma'
-        )
+            expected_name = 'CSV import to scratch.api_client_test_fixture'
+            expected_kwargs = {
+                'name': expected_name,
+                'max_errors': None,
+                'existing_table_rows': 'truncate',
+                'hidden': True,
+                'column_delimiter': 'comma',
+                'compression': 'gzip',
+                'escaped': False,
+                'execution': 'immediate',
+                'loosen_types': False,
+                'table_columns': table_columns,
+                'redshift_destination_options': {
+                    'diststyle': None, 'distkey': None,
+                    'sortkeys': [None, None]
+                }
 
-        expected_name = 'CSV import to scratch.api_client_test_fixture'
-        expected_kwargs = {
-            'name': expected_name,
-            'max_errors': None,
-            'existing_table_rows': 'truncate',
-            'hidden': True,
-            'column_delimiter': 'comma',
-            'compression': 'gzip',
-            'escaped': False,
-            'execution': 'immediate',
-            'loosen_types': False,
-            'table_columns': table_columns,
-            'redshift_destination_options': {
-                'diststyle': None, 'distkey': None,
-                'sortkeys': [None, None]
             }
+            self.mock_client.imports.post_files_csv.assert_called_once_with(
+                {'file_ids': mock_cleaned_file_ids},
+                {
+                    'schema': 'scratch',
+                    'table': 'api_client_test_fixture',
+                    'remote_host_id': 42,
+                    'credential_id': 713,
+                    'primary_keys': None,
+                    'last_modified_keys': None
+                },
+                True,
+                **expected_kwargs
+            )
 
-        }
-        self.mock_client.imports.post_files_csv.assert_called_once_with(
-            {'file_ids': [mock_cleaned_file_id]},
-            {
-                'schema': 'scratch',
-                'table': 'api_client_test_fixture',
-                'remote_host_id': 42,
-                'credential_id': 713,
-                'primary_keys': None,
-                'last_modified_keys': None
-            },
-            True,
-            **expected_kwargs
-        )
+        # Check that things work with a single file ID or multiple IDs.
+        # In particular, we want to ensure that loosen_types is set to False
+        # in both situations.
+        for mock_file_ids in (1234, [1234], [1234, 1235]):
```

Comment:
I'm more used to using `pytest.mark.parametrize` for this sort of thing, but that seems to not be possible since this is a method of a `unittest.TestCase`, so I tried to follow [this SO answer](https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python).  The `subTest` context manager isn't super necessary, but I think it might improve the error messages a bit when tests fail.